tests/unit/misc/test_cmdhistory.py

```diff
@@ -0,0 +1,196 @@
+                                         HistoryEndReachedError)
+
+
+HISTORY = ['first', 'second', 'third', 'fourth', 'fifth']
+
+CONFIG_NOT_PRIVATE = {'general': {'private-browsing': False}}
+CONFIG_PRIVATE = {'general': {'private-browsing': True}}
+
+
+class TestConstructor:
+
+    """Tests for the constructor."""
+
+    def test_no_history(self):
+        hist = History()
+        # .history should equal []
+        assert len(hist.history) == 0
+
+    def test_history(self):
+        hist = History(history=HISTORY)
+        assert hist.history == HISTORY
+
+
+class TestCommandHistory:
+
+    """Tests for Command History."""
+
+    @pytest.fixture(autouse=True)
+    def setup(self):
+        self.hist = History(history=HISTORY)
+
+    def test_is_browsing(self):
+        """Test is_browsing()."""
+
+        self.hist._tmphist = None
+        assert not self.hist.is_browsing()
+
+        self.hist._tmphist = HISTORY
+        assert self.hist.is_browsing()
+
+    def test_start_stop(self):
+        """Test the start/stop."""
+
+        # We can use is_browsing() because it is tested above
+        assert not self.hist.is_browsing()
+        self.hist.start('s')
+        assert self.hist.is_browsing()
+        self.hist.stop()
+        assert not self.hist.is_browsing()
+
+    def test_start_with_text(self):
+        """Test start with given 'text'."""
+        self.hist.start('f')
+        assert 'first' in self.hist._tmphist
+        assert 'fourth' in self.hist._tmphist
+        assert 'second' not in self.hist._tmphist
+
+    def test_start_no_text(self):
+        """Test start with no given text."""
+        self.hist.start('')
+
+        # There is probably a better way for NeighbourList?
+        for i in self.hist._tmphist:
+            assert i in HISTORY
+
+        for i in HISTORY:
+            assert i in self.hist._tmphist
+
+    def test_start_no_items(self):
+        """Test start with no matching text."""
+        with pytest.raises(HistoryEmptyError) as excinfo:
+            self.hist.start('k')
+        assert str(excinfo.value) == "History is empty."
+        assert not self.hist._tmphist
+
+    def test_get_item(self):
+        """Test __get_item__."""
+        for i in range(0, len(HISTORY)):
+            assert self.hist[i] == HISTORY[i]
+
+    def test_not_browsing_error(self):
+        """Test that next/previtem throws a ValueError"""
+        with pytest.raises(ValueError) as error1:
+            self.hist.nextitem()
+        assert str(error1.value) == "Currently not browsing history"
+
+        with pytest.raises(ValueError) as error2:
+            self.hist.previtem()
+        assert str(error2.value) == "Currently not browsing history"
+
+    def return_item(self):
+        return 'item'
+
+    def test_nextitem_single(self):
+        """Test nextitem() with valid input."""
+        self.hist.start('f')
+        self.hist._tmphist.nextitem = self.return_item
```

Comment:
This is another place where I think it'd make sense to use `... = lambda: 'item'` instead of defining a function. Even though it's used in different places, I think it's so simple that it doesn't hurt to repeat it.